design/status-design-update.md

```diff
@@ -0,0 +1,1166 @@
+
+Required:
+
+- Detail expected personas for users
+- Provide a mechanism for detailed information about invalid inclusions, including specifying what is added to each object, includer and included.
+- Provide a mechanism for detailed information about invalid routes/services.
+- Provide a mechanism for HTTPProxy administrators to see what conditions are added at each level of inclusion.
+- Provide the information required for higher-level parsing of the status (to enable both kubectl and other UI inspection).
+- An external tool must be able to build a representation of the HTTPProxy include DAG without speaking using only the Kubernetes API.
+- Provide a set of examples that can be used both for understanding this proposal and as the basis for a test suite for status.
+
+Optional:
+
+- Add usable printer columns for kubectl use (Fixes #1567)
+
+### Non-Goals
+
+- Considering how this will be implemented at anthing but the most basic level.
+
+This is a high-level design effort, intended to think about what the ideal case would be.
+Let's understand the dream first, then work back from there.
+
+## Users/Personas
+
+### Proxy Administrator
+
+This is the person or team who controls the TLD and any associated TLS information.
+They are responsible for root HTTPProxy objects, and specifying conditions on the includes for the root HTTPProy objects.
+
+### Service Administrator
+
+This is the person or team who controls a HTTPProxy that is included into another.
+Similarly to HTTPProxy inclusion, there may be an arbitrarily long chain of Service Administrators before an actual service is reached.
+
+### Upstream tool user
+
+This is the person or team who wants to be able to visualize the overall set of HTTPProxy objects, how they interact, and get a picture of what the eventual Envoy config will look like.
+This user only wants to use the Kubernetes API to retrieve this information.
+
+Note that Proxy admins and Service admins may also be Upstream tool users.
+
+## Overall Design
+
+The overall design I have in mind is to add some further fields to the status object in each HTTPProxy, and allow for more information about the in-between states, other than 'everything is okay' and 'nothing is okay'.
+
+Each one of these fields is intended to include enough information to enable either troubleshooting of specific errors, to expose information for higher-level tooling, or both.
+
+### Extra fields
+
+There are three extra fields I believe we must add, and one I believe we should.
+The last is only optional only because I think that we could approximate its functions using the other fields.
+I think that the design is cleaner with it included, however.
+
+For each place where we add a status, I think we should keep the current naming scheme:
+
+- `currentStatus` is exactly what it says - the current status. This should have three values, `Valid`, `Invalid`, and `Warning` (or similar).
```

Comment:
What about `orphaned` which is neither valid or invalid.

Reply:
Good point!

Reply:
What happens if a HTTPProxy root is included by another HTTPProxy root -- is it valid or invalid? The first path, root -> root, the parent is invalid, but is the child invalid because the parent made a boo boo?

Reply:
In my mind, that would be a 'Warning' maybe? It's a good example though.